gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -78,6 +86,36 @@
     this.showUnsavedChangesDialogEvents = new SingleLiveEvent<>();
     this.showErrorDialogEvents = new SingleLiveEvent<>();
     this.authManager = authenticationManager;
+    this.argumentProcessor = BehaviorProcessor.create();
+    this.recordProcessor = BehaviorProcessor.create();
+
+    Flowable<Resource<Record>> saveRecordStream =
+        recordProcessor.flatMap(
+            r ->
+                this.authManager
+                    .getUser()
+                    .flatMap(user -> this.dataRepository.saveChanges(r, getChangeList(r), user))
+                    .toFlowable(BackpressureStrategy.LATEST));
```

Comment:
I think `switchMap` would also be more correct here (see other comment re the same).